nodes/generators_extended/hypercube.py

```diff
@@ -0,0 +1,253 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import IntProperty, FloatProperty, BoolProperty, EnumProperty
+
+from math import sin, cos, pi, sqrt, radians
+from random import random
+import time
+from mathutils import Vector, Matrix
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode, match_long_repeat
+
+# dictionary to store once the unit/untransformed hypercube verts, edges & polys
+_hypercube = {}
+
+
+def project(vert4D, d):
+    '''
+        Project a 4D vector onto 3D space.
+    '''
+    cx, cy, cz = [0.0, 0.0, 0.0]  # center
+    x, y, z, t = vert4D
+    return [x + (cx - x)*t/d, y + (cy - y)*t/d, z + (cz - z)*t/d]
```

Comment:
well, `d` seems to me like it will always be `0` at least once (at the start of iterating through `range(len(verts4D))`.  if len(verts4D) is larger than 0.

how do you know the bug is the result of a _memory leak_? does it give an error.. or crash blender?  i'm too occupied to try.